crypto/cryptlib.c

```diff
@@ -19,29 +19,91 @@
 extern unsigned int OPENSSL_ia32cap_P[4];
 
 # if defined(OPENSSL_CPUID_OBJ) && !defined(OPENSSL_NO_ASM) && !defined(I386_ONLY)
-#include <stdio.h>
+
+/*
+ * Purpose of these minimalistic and character-type-agnostic subroutines
+ * is to break dependency on MSCVRT (on Windows) and locale. This makes
+ * OPENSSL_cpuid_setup safe to use as "constructor". "Character-type-
+ * agnostic" means that they work with either wide or 8-bit characters,
+ * exploiting the fact that first 127 characters can be simply casted
+ * between the sets, while the rest would be simply rejected by ossl_is*
+ * subroutines.
```

Comment:
I'm not sure how this is supposed to work. You get a char *. Every 2 characters will be a \0 if it's UTF-16

Reply:
I must have somehow missed the ossl_char

Reply:
How about renaming ossl_char to variant_char? Hopefully it would be harder to miss then...